tests/system/Validation/ValidationTest.php

```diff
@@ -956,4 +957,110 @@ public function testTranslatedLabelTagReplacement()
 	}
 
 	//--------------------------------------------------------------------
+
+	/**
+	 * internal method to simplify placeholder replacement test
+	 * REQUIRES THE RULES TO BE SET FOR THE FIELD "foo"
+	 *
+	 * @param array|null $data optional POST data, needs to contain the key $placeholderField to pass
+	 */
+	protected function testPlaceholderReplacement(string $placeholder = 'id', ?array $data = NULL) {
```

Comment:
if this is internal then don't name it with `test*` to not confuse test methods.